### PR TITLE
Examples and tutorials for TensorFlow optimization.

### PR DESCRIPTION
**Pull request by JamesTheZ**
1. Examples of BladeDISC optimization for TensorFlow Inference and Training.
2. Tutorial for TensorFlow inference/training optimization with BladeDISC.

### CODE REVIEW

File: docs/tutorials/tensorflow_inference_and_training.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+The above code shows the complete process to optimize BERT inference model with BladeDISC.
+Please refer to [TensorFlow BERT Inference Example](/examples/TensorFlow/Inference/BERT) 
+for more scripts to compare the performance of BladeDISC optimization with naive TensorFlow and XLA.
+
+
+## DeePMD training
+
+We take a deep learning based model for molecular dynamics (MD) to show how to optimize a training model with BladeDISC.
```
Comment (linearhit) [inline docs/tutorials/tensorflow_inference_and_training.md:95]: The length of each line should be limited for a md file. You can use a 'rewrap' plugin of Vscode to reformat the file.

**Review by linearhit — CHANGES_REQUESTED** [review]

File: docs/tutorials/tensorflow_inference_and_training.md (hunk 1 above)
Comment (JamesTheZ) [reply docs/tutorials/tensorflow_inference_and_training.md]: Done. Thanks.

File: docs/tutorials/tensorflow_inference_and_training.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# Tutorial: Optimize TensorFlow Models with BladeDISC
+
+In this tutorial, we show how to optimize TensorFlow models with BladeDISC for
+both inference and training. Users only need to add two lines of code to
+optimize the model just-in-time for the examples in this tutorial.
+
+These packages are required before going through the tour:
```
Comment (Yancey1989) [inline docs/tutorials/tensorflow_inference_and_training.md:7]: Users don't need to install these two packages if they have installed BladeDISC as `Install BladeDISC With Docker` docs.

File: docs/tutorials/tensorflow_inference_and_training.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- [DeePMD training](#deepmd-training)
+  - [Prologue: install DeePMD-kit and download
+    data](#prologue-install-deepmd-kit-and-download-data)
+  - [Still, all you need are the two
+    lines!](#still-all-you-need-are-the-two-lines)
+  - [Epilogue: normal process to run MD training with DeePMD-kit
+    API](#epilogue-normal-process-to-run-md-training-with-deepmd-kit-api)
+
+
+## BERT inference
```
Comment (Yancey1989) [inline docs/tutorials/tensorflow_inference_and_training.md:31]: BERT Inference

File: docs/tutorials/tensorflow_inference_and_training.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+BERT models are usually feed with data of dynamic shapes in real production. The
+dynamic shape mainly comes from two aspects, one is varied batch-size, and the
+other is varied data shape of each sample (e.g., varied sequence lengths). The
+model we show in this tutorial has varied batch-size.
+
+### Prologue: download frozen model
```
Comment (Yancey1989) [inline docs/tutorials/tensorflow_inference_and_training.md:38]: Please format the title using https://titlecase.com/

File: docs/tutorials/tensorflow_inference_and_training.md (hunk 2 above)
Comment (JamesTheZ) [reply docs/tutorials/tensorflow_inference_and_training.md]: Done. Thanks.

File: docs/tutorials/tensorflow_inference_and_training.md (hunk 3 above)
Comment (JamesTheZ) [reply docs/tutorials/tensorflow_inference_and_training.md]: Done. Thanks.

File: docs/tutorials/tensorflow_inference_and_training.md (hunk 4 above)
Comment (JamesTheZ) [reply docs/tutorials/tensorflow_inference_and_training.md]: Done. Thanks.

**Review by linearhit — APPROVED** [review]
LGTM

**Pull request merged**